xdsl/transforms/stencil_to_csl_stencil.py

```diff
@@ -1,23 +1,145 @@
 from collections.abc import Sequence
-
-from attr import dataclass
+from dataclasses import dataclass
+from math import prod
 
 from xdsl.context import MLContext
-from xdsl.dialects import memref, stencil, tensor
-from xdsl.dialects.builtin import ModuleOp, TensorType
+from xdsl.dialects import arith, memref, stencil, tensor
+from xdsl.dialects.builtin import (
+    IndexType,
+    IntegerAttr,
+    IntegerType,
+    ModuleOp,
+    TensorType,
+)
 from xdsl.dialects.csl import csl_stencil
 from xdsl.dialects.experimental import dmp
-from xdsl.ir import Attribute, OpResult
+from xdsl.ir import Attribute, Block, BlockArgument, Operation, OpResult, Region
+from xdsl.irdl import Operand
 from xdsl.passes import ModulePass
 from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
     PatternRewriter,
     PatternRewriteWalker,
     RewritePattern,
     op_type_rewrite_pattern,
 )
+from xdsl.rewriter import InsertPoint
+from xdsl.transforms.experimental.stencil_tensorize_z_dimension import (
+    BackpropagateStencilShapes,
+)
 from xdsl.utils.hints import isa
 
 
+def get_stencil_accessed_symbols(op: Operand) -> set[Operand]:
```

Comment:
Maybe something closer to

```suggestion
def find_transitive_stencil_access_operands(op: Operand) -> set[Operand]:
```

Reply:
Changed to `get_stencil_access_operands`